phpBB/phpbb/textreparser/base.php

```diff
@@ -0,0 +1,211 @@
+			$record[$field_name] = $this->guess_bbcode($record, $bbcode);
+		}
+
+		// Magic URLs are tied to the URL BBCode, that's why if magic URLs are enabled we make sure
+		// that the URL BBCode is also enabled
+		if ($record['enable_magic_url'])
+		{
+			$record['enable_url_bbcode'] = true;
+		}
+
+		return $record;
+	}
+
+	/**
+	* Guess whether given BBCode is in use in given record
+	*
+	* @param  array  $record
+	* @param  string $bbcode
+	* @return bool
+	*/
+	protected function guess_bbcode(array $record, $bbcode)
+	{
+		if (!empty($record['bbcode_uid']))
+		{
+			// Look for the closing tag, e.g. [/url]
+			$match = '[/' . $bbcode . ':' . $record['bbcode_uid'];
+			if (strpos($record['text'], $match) !== false)
+			{
+				return true;
+			}
+		}
+
+		if (substr($record['text'], 0, 2) == '<r')
+		{
+			// Look for the closing tag inside of a e element, in an element of the same name, e.g.
+			// <e>[/url]</e></URL>
+			$match = '<e>[/' . $bbcode . ']</e></' . strtoupper($bbcode) . '>';
+			if (strpos($record['text'], $match) !== false)
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	* Guess whether any BBCode is in use in given record
+	*
+	* @param  array $record
+	* @return bool
+	*/
+	protected function guess_bbcodes(array $record)
+	{
+		if (!empty($record['bbcode_uid']))
+		{
+			// Test whether the bbcode_uid is in use
+			$match = ':' . $record['bbcode_uid'];
+			if (strpos($record['text'], $match) !== false)
+			{
+				return true;
+			}
+		}
+
+		if (substr($record['text'], 0, 2) == '<r')
+		{
+			// Look for a closing tag inside of an e element
+			return (bool) preg_match('(<e>\\[/\\w+\\]</e>)', $match);
+		}
+
+		return false;
+	}
+
+	/**
+	* Guess whether magic URLs are in use in given record
+	*
+	* @param  array $record
+	* @return bool
+	*/
+	protected function guess_magic_url(array $record)
+	{
+		// Look for <!-- m --> or for a URL tag that's not immediately followed by <s>
+		return (strpos($record['text'], '<!-- m -->') !== false || preg_match('(<URL [^>]++>(?!<s>))', $record['text']));
+	}
+
+	/**
+	* Guess whether smilies are in use in given record
+	*
+	* @param  array $record
+	* @return bool
+	*/
+	protected function guess_smilies(array $record)
+	{
+		return (strpos($record['text'], '<!-- s') !== false || strpos($record['text'], '<E>') !== false);
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function reparse_range($min_id, $max_id)
+	{
+		foreach ($this->get_records_by_range($min_id, $max_id) as $record)
+		{
+			$this->reparse_record($record);
+		}
+	}
+
+	/**
+	* Reparse given record
+	*
+	* @param  array $record Associative array containing the record's data
+	*/
+	protected function reparse_record(array $record)
+	{
+		$record = $this->add_missing_fields($record);
+		$unparsed = array_merge(
+			$record,
+			generate_text_for_edit(
+				$record['text'],
+				$record['bbcode_uid'],
+				OPTION_FLAG_BBCODE | OPTION_FLAG_SMILIES | OPTION_FLAG_LINKS
```

Comment:
As far as I know, the third argument to `generate_text_for_edit()` has no effect on anything. Its value is passed to extensions and determines the return values of `allow_bbcode`, `allow_smilies` and `allow_urls` but those are not used for anything. The settings for toggling features are prefixed with `enable_`.

I could set it to `0` and it wouldn't change anything. There are 8 posts in the `post_text` test fixture that represent every combination of the 3 settings and the third argument to `generate_text_for_edit()` does not affect the test's result.

Come to think of it, I will just generate a value based on the `enable_` field so that nobody else has to wonder about it. That will be faster that explaining that the value doesn't matter.